network/builder/src/builder.rs

```diff
@@ -108,12 +112,16 @@ pub struct NetworkBuilder {
 
     connectivity_manager_config: Option<ConnectivityManagerBuilderConfig>,
     connectivity_manager: Option<ConnectivityManagerService>,
+    configuration_change_listener_cfg: Option<ConfigurationChangeListenerConfig>,
+    configuration_change_listener: Option<ConfigurationChangeListener>,
     discovery_cfg: Option<DiscoveryBuilderConfig>,
     discovery: Option<DiscoveryService>,
     health_checker_cfg: Option<HealthCheckerBuilderConfig>,
     health_checker: Option<HealthCheckerService>,
     onchain_discovery_cfg: Option<OnchainDiscoveryBuilderConfig>,
     onchain_discovery: Option<OnchainDiscoveryBuilder>,
+
+    reconfig_subscriptions: Vec<ReconfigSubscription>,
```

Comment:
why not put this in ConfigurationChangeListener

Reply:
The reconfig subscriptions are needed (later on) by StateSync, so network_builder must provide an accessing interface.    We can implement that either by invoking all internal components that produce subscriptions or by pro-actively gathering the subscriptions as they are generated.

I chose the latter, but there is no fundamental reason to not choose the former.

Reply:
I see, given that we have these other builders and it is only used once, should we push it into the builders themselves?

Reply:
I lean towards keeping this specifically at the network_builder level because it is somehow 'specific' to StateSync.  So it is good for network_builder to consolidate things together since it can, in principle, come from multiple internal components.

There are other things, such as conn_mgr_reqs_tx, which should remain with the "internal" component because multiple external components may access it.

That is actually the principled distinction that I draw:

- For wires that are many-external to one-internal, the wire should live internally.
- For wires that are one-external to **-internal, the wires should leave as close to the external boundary as possible.